pkg/controller/node/nodecontroller.go

```diff
@@ -644,16 +654,31 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
-func nodeExistsInCloudProvider(cloud cloudprovider.Interface, nodeName string) (bool, error) {
-	instances, ok := cloud.Instances()
-	if !ok {
-		return false, fmt.Errorf("%v", ErrCloudInstance)
-	}
-	if _, err := instances.ExternalID(nodeName); err != nil {
-		if err == cloudprovider.InstanceNotFound {
-			return false, nil
+type nodeExistanceChecker struct {
+	lock           sync.Mutex
+	instances      sets.String
+	lastUpdateTime time.Time
+	refreshPeriod  time.Duration
+}
+
+func (c *nodeExistanceChecker) nodeExistsInCloudProvider(cloud cloudprovider.Interface, nodeName string) (bool, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	now := time.Now()
+	if c.instances == nil || now.After(c.lastUpdateTime.Add(c.refreshPeriod)) {
+		instances, ok := cloud.Instances()
+		if !ok {
+			return false, fmt.Errorf("%v", ErrCloudInstance)
 		}
-		return false, err
+		instanceNames, err := instances.List("")
+		if err != nil {
+			return false, err
+		}
+		c.instances = sets.NewString(instanceNames...)
```

Comment:
I know that you think it will help with #26170, but I'm not convinced. So I would wait for the confirmation that nodecontroller is really a victim here, before merging something like this.